optuna/integration/mlflow.py

```diff
@@ -1,20 +1,149 @@
+
+        Args:
+            study (optuna.study.Study): Study to be tracked in MLFlow
+        """
+        # This sets the tracking_uri for MLflow.
+        if self._tracking_uri is not None:
+            mlflow.set_tracking_uri(self._tracking_uri)
+
+        # This sets the experiment of MLflow.
+        mlflow.set_experiment(study.study_name)
+
+    def set_tags(self, trial: optuna.trial.BaseTrial, study: optuna.study.Study) -> None:
+        """Sets the Optuna tags for the current MLFlow run
+
+        Args:
+            trial (optuna.trial.BaseTrial): Trial to be tracked
+            study (optuna.study.Study): Study to be tracked
+        """
+        tags: Dict[str, str] = {}
+        tags["number"] = str(trial.number)
+        tags["datetime_start"] = str(trial.datetime_start)
+        if isinstance(trial, optuna.trial.FrozenTrial):
+            tags["datetime_complete"] = str(trial.datetime_complete)
+
+            # Set state and convert it to str and remove the common prefix.
+            trial_state = trial.state
+            if isinstance(trial_state, TrialState) and trial_state.is_finished():
+                tags["state"] = str(trial_state).split(".")[-1]
+
+        # Set direction and convert it to str and remove the common prefix.
+        study_direction = study.direction
+        if isinstance(study_direction, StudyDirection):
+            tags["direction"] = str(study_direction).split(".")[-1]
+
+        tags.update(trial.user_attrs)
+        distributions = {(k + "_distribution"): str(v) for (k, v) in trial.distributions.items()}
+        tags.update(distributions)
+
+        if self._tag_study_user_attrs:
+            tags.update(study.user_attrs)
+
+        # This is a temporary fix on Optuna side. It avoids an error with user
+        # attributes that are too long. It should be fixed on MLflow side later.
+        # When it is fixed on MLflow side this codeblock can be removed.
+        # see https://github.com/optuna/optuna/issues/1340
+        # see https://github.com/mlflow/mlflow/issues/2931
+        max_mlflow_tag_length = 5000
+        for key, value in tags.items():
+            value = str(value)  # make sure it is a string
+            if len(value) > max_mlflow_tag_length:
+                tags[key] = textwrap.shorten(value, max_mlflow_tag_length)
+
+        # This sets the tags for MLflow.
+        mlflow.set_tags(tags)
+
+    def log_metric(self, value: Union[float, Sequence[float], None]) -> None:
+        """Log the trial result as metric to MLFlow
+
+        Args:
+            value (Union[float, Sequence[float], None]): Result of trial
+        """
+        if isinstance(value, Sequence):
+            for key, v in zip(self._metric_name, value):
+                mlflow.log_metric(key, v)
+        else:
+            trial_value = value if value is not None else float("nan")
+            mlflow.log_metric(self._metric_name, trial_value)
```

Comment:
As you mentioned later on, the multi objective support needs additional changes and I agree that it should be moved to an other PR. Since this part is only used for logging the objective value and all other metrics should be logged in the objective function itself, there is no need for having the sequence yet.